### PR TITLE
Bring branch up to date, begin 2.18 updates

### PR DESCRIPTION
**Pull request by MaryaBelanger**
v2.18 branch was behind main by 8 commits, pulled those in.

also began initial changes for 2.18 updates:
* added 2.18 section to What's new page and included link to change log
* added 2.18 section to Language evolution page

(mostly just testing my git config/fork/branch/commit/push setup)

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
 
       # Initializes the CodeQL tools for scanning.
       - name: Initialize CodeQL
-        uses: github/codeql-action/init@2ca79b6fa8d3ec278944088b4aa5f46912db5d63
+        uses: github/codeql-action/init@f5d217be74900c6ac8fbbe53f3c10376ba4e64da
```
Comment (domesticmouse) [inline .github/workflows/codeql-analysis.yml:36]: I'm curious why this change is included in this PR. https://github.com/dart-lang/site-www/pull/4180 landed this change already, IIUC.

@parlough any ideas?

File: Dockerfile [hunk 2]
```diff
@@ -80,7 +80,7 @@ CMD ["./tool/test.sh"]
 FROM dart as node
 RUN set -eu; \
     NODE_PPA="node_ppa.sh"; \
-    NODE_SHA256=9820c0fcf01527ffd3b2077de1f76d4bbe67bdb38df9d12fa195d7eea1521e8a; \
+    NODE_SHA256=27932797347f900242caaaeba5c1d7c965b3da70566d81123b15be1c0b80cc2c; \
```
Comment (domesticmouse) [inline Dockerfile:83]: Also, this smells odd. 

PTAL @parlough

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (parlough) [reply .github/workflows/codeql-analysis.yml:36]: I assume you noticed, but this is a separate branch tracking 2.18.

Comment (domesticmouse) [reply .github/workflows/codeql-analysis.yml:36]: > I assume you noticed, but this is a separate branch tracking 2.18. This is just merging those new commits in.

Yeah i figured it out finally =)

Comment (MaryaBelanger) [reply .github/workflows/codeql-analysis.yml:36]: Ahh, sorry all, I was refreshing myself on git. You can ignore this for the most part, it's just bringing this branch up to speed with main (my commits will be moved to a different pr)
